CMSgov/easi-app
# [NOREF] Refactor a few TRB Status Calculations

# NOREF

## Changes and Description

- Refactored a few TRB Status calculations by using the already in-memory reference to the TRB Request (on the resolver) instead of querying the DB for it multiple times
- Add to `trb_request_status_test.go` to include tests for the task list section statuses, rather than just the overall statuses

## How to test this change

- Start the app `scripts/dev up`
- Run automated backend tests `scripts/dev test:go`
- Clean & Seed the database `scripts/dev db:clean && scripts/dev db:seed`
- Open the app, login (local is fine) as `ABCD`, click through each seeded request
- Make sure task list statuses and overall statuses of requests appear as expected

## PR Author Review Checklist



## PR Reviewer Guidelines
- It's best to pull the branch locally and test it, rather than just looking at the code online!
- Check that all code is adequately covered by tests - if it isn't feel free to suggest the addition of tests.
- Always make comments, even if it's minor, or if you don't understand why something was done.
